src/main/kotlin/tech/gdragon/BotUtils.kt

```diff
@@ -266,25 +265,40 @@ object BotUtils {
    * In the past, I've been deleting the Guild from the database, but that makes things annoying when you rejoin.
    * For now, we'll just be leaving a Guild, but keeping the settings.
    */
-  fun leaveAncientGuilds(jda: JDA, afterDays: Int) {
+  fun leaveInactiveGuilds(jda: JDA, afterDays: Int, whitelist: List<Long>) {
     logger.info { "Leaving all Guilds that haven't been active in the past $afterDays days." }
     val op: SqlExpressionBuilder.() -> Op<Boolean> = {
       val now = DateTime.now()
-      not(Between(Tables.Guilds.lastActiveOn, dateTimeLiteral(now.minusDays(afterDays)), dateTimeLiteral(now)))
+      val from = now.minusDays(afterDays)
+      val inactiveGuildOp = Guilds.active.eq(true).and(Guilds.lastActiveOn.less(from))
+
+      if (whitelist.isNotEmpty()) {
+        val whitelistEntityIDs = whitelist.map { EntityID(it, Guilds) }
+        inactiveGuildOp.and(Guilds.id.notInList(whitelistEntityIDs))
+      } else {
+        inactiveGuildOp
+      }
     }
 
-    // Find all ancient guilds and ask the Bot to leave them
+    // Find all ancient guilds and ask the Bot to leave them, or mark them as inactive if already gone
     transaction {
-      Guild.find(op).toList()
-    }.forEach {
-      val guild = jda.getGuildById(it.id.value)
-      guild
-        ?.leave()
-        ?.queue({
-          logger.info { "Left server '$guild', reached inactivity period." }
-        }, { e ->
-          logger.error(e) { "Could not leave server '$guild'!" }
-        })
+      val guilds = Guild.find(op).toList()
+
+      guilds
+        .forEach {
+          val guild = jda.getGuildById(it.id.value)
+          guild
+            ?.leave()
+            ?.queue({
+              BotUtils.logger.info { "Left server '$guild', reached inactivity period." }
+            }, { e ->
+              BotUtils.logger.error(e) { "Could not leave server '$guild'!" }
+            })
+            ?: BotUtils.logger.warn {
+              it.active = false
+              "No longer in this guild ${it.name}, but marking as inactive"
+            }
```

Comment:
If the bot isn't in an inactive guild, `guild?.leave()` would silently be skipped but the guild would still be marked as active in the database, this should address and fix prior problems.